### DISCUSSION TITLE
Add setup and teardown code for `supabase db test`

### DISCUSSION THREAD

**Discussion by isaacharrisholt** [supabase/supabase#13053] [Feature Requests]
### The problem

When running tests with `supabase db test`, it's useful to be able to run SQL as an auth user as described in [the docs for RLS](https://supabase.com/docs/guides/auth/row-level-security#testing-policies).

We don't want those functions to be in the database permanently for security reasons, so it would be good if they could be created at the start of the test then town down at the end.

### Proposed solution

`supabase db test` could look for `setup.sql` and `teardown.sql` files in the `supabase/tests/database` directory and run them before and after the tests respectively. 

Currently, we have a procedure that creates these functions, and another to delete them, but it's a pain to have to call those at the start and end of each test.

**Comment by mrmikardo** [comment]
I'd like this too!

**Comment by silentworks** [comment]
It probably won't be a setup and teardown as such but we will allow you to include additional file which you can then write your functions in while putting them inside and since your tests rollback at the end they won't stay in the DB after the test ends.

**Reply by mrmikardo** [reply]
Is there a PR somewhere for this? (Is work to implement this functionality already in-progress).

**Reply by silentworks** [reply]
No this work is not in progress as yet. But it is on the radar.
